### PR TITLE
COV4 support in asm

### PR DESCRIPTION
**Pull request by shurale-nkn**
Added macro for unified metadata creation

### CODE REVIEW

File: src/kernels/target_id.inc [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+.macro send_TARGET_ID_str next_m
+    \next_m amdgcn-amd-amdhsa--\()\proc\xnack_str\sramecc
+.endm //send_TARGET_ID_str
+
+.endm //create_TARGET_ID_with_sramecc
+.endm //create_TARGET_ID_with_xnack
+.endm //create_TARGET_ID_with_proc
+
+.macro create_TARGET_ID_xnack_status
+    .if(amd_target_feature_xnack == 2)
```
Comment (atamazov) [inline src/kernels/target_id.inc:59]: I thought we wanted to make it so:
- 0 - OFF
- 1 - ON
- 2 - don't care

File: src/kernels/target_id.inc [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+////////////////////////////////
+// !!!! Before using, required to define xnack and sramecc suport in amd_target_feature_xnack and amd_target_feature_sramecc variables.
```
Comment (atamazov) [inline src/kernels/target_id.inc:28]: Can we check that programmatically?

Comment (atamazov) [inline src/kernels/target_id.inc:28]:
```suggestion
// THE FOLLOWING SYMBOLS MUST BE DEFINED PRIOR USING THE MACROS DECLARED BELOW:
// * amd_target_feature_xnack
// * amd_target_feature_sramecc
// Allowed values: 
// * 0 - The feature is OFF on the target device
// * 1 - The feature is ON
// * 2 - The target device does not have the feature OR the kernel does not depend on its state.
```

**Review by atamazov — COMMENTED** [review]
Some superfluous comments

**Comment by daniellowell** [comment]
This PR has issues in our CI:
```
<stdin>:10624:1: error: Undefined amd_target_feature_xnack
.error "Undefined amd_target_feature_xnack"
^
<stdin>:13307:1: error: unmatched .ifs or .elses
^
```

**Comment by shurale-nkn** [comment]
@carlushuang 
commit cadf08f522f70cfb contains macro`igemm_amdgpu_metadata` for igemm kernels.
It automatically adds the lines required in COV4 to the amdgpu_metadata section. 
If you update the igemm generator to use this macro, it will be easier to update metadata in many kernels in the future.

**Comment by carlushuang** [comment]
@shurale-nkn thanks, I see the macros. I'll add them into the generator to let future update smooth.

**Comment by atamazov** [comment]
Converted to draft to avoid spoiling docker image cache on CI. Please change this back to "normal" PR only after ending of CI moratorium (see mail). Then first merge from develop. Push the `[Ready for review]` button only after that. Thanks.

**Comment by junliume** [comment]
@shurale-nkn  any plan to continue on this PR or we can close it?

**Comment by shurale-nkn** [comment]
> @shurale-nkn any plan to continue on this PR or we can close it?
@junliume 
It seems that the problem has already lost its relevance, please delete it

**Comment by atamazov** [comment]
@shurale-nkn @junliume What we will do if importance dramatically rise tomorrow?

**Comment by JehandadKhan** [comment]
@shurale-nkn  Can we close this PR if it has been abandoned.

**Pull request closed without merging**